src/gsy_e/models/state.py

```diff
@@ -381,6 +385,12 @@ def _calculate_unsettled_energy_kWh(
                              self.get_available_energy_kWh(time_slot))
         return traded_energy_kWh - measured_energy_kWh
 
+    def get_results_dict(self, current_time_slot: DateTime) -> dict:
+        return {
+            "pv_production_kWh": self.get_energy_production_forecast_kWh(current_time_slot),
+            "available_energy_kWh": self.get_available_energy_kWh(current_time_slot)
+        }
```

Comment:
This looks quite similar to the `get_state()` method already implemented in the parent class (e.g. `ProductionState`). Do you think there could be a way to re-use that? The reason why I'm focusing on this is that I'd like to use `get_state()` as the only source of truth for all the "global recap" information about the `State` class. So that if we change `get_state()` this will also be reflected automatically in the `get_results_dict()` method without any additional effort.

Reply:
I don't think this is a good idea conceptually. The state of an object is in general way different from the parts of it that are exposed to the user (which is actually the simulation results). A prime example is the StorageState, where the internal state includes many parameters, but the results demand only the SOC. 
The alternative would be to just pass the state and do the filtering on the simulation results parsing, but this is a reactive approach that I do not like. IMO it is much better for the class itself to have different functions that report parts of the internal state for the different needed usecases.

Reply:
Ok, if the two states should be different then I agree 👍